modules/broadcast/src/backend/daemon.ts

```diff
@@ -92,14 +93,12 @@ export default async (botId: string, bp: SDK, db: Database) => {
     })
 
     await db.updateErrorField(scheduleId)
-
     await db.deleteBroadcastOutboxById(scheduleId)
   }
 
-  async function scheduleToOutbox(botId) {
-    const schedulingLock = await bp.distributed.acquireLock('broadcast/lock/scheduling', ms('5m'))
-
-    if (!db.knex || !schedulingLock) {
+  const scheduleToOutbox = async (botId: string) => {
+    const schedulingLock = await bp.distributed.acquireLock(`broadcast/lock/scheduling_${botId}`, ms('5m'))
```

Comment:
Logic is a single node from the cluster sends all scheduled outbox every 5 minutes ?

Reply:
First refactor is to do a small cleanup, it will definitely need a second one to reorganize some of the stuff